controller/network/route_perf_test.go

```diff
@@ -0,0 +1,173 @@
+package network
+
+import (
+	"fmt"
+	"github.com/michaelquigley/pfxlog"
+	"github.com/openziti/fabric/controller/db"
+	"github.com/openziti/foundation/identity/identity"
+	"github.com/openziti/foundation/transport/tcp"
+	"github.com/sirupsen/logrus"
+	"math/rand"
+	"testing"
+)
+
+func BenchmarkShortestPathPerf(b *testing.B) {
+	b.StopTimer()
+	pfxlog.GlobalInit(logrus.WarnLevel, pfxlog.DefaultOptions())
+
+	ctx := db.NewTestContext(b)
+	defer ctx.Cleanup()
+
+	closeNotify := make(chan struct{})
+	defer close(closeNotify)
+
+	network, err := NewNetwork("test", nil, ctx.GetDb(), nil, NewVersionProviderTest(), closeNotify)
+	ctx.NoError(err)
+
+	addr := "tcp:0.0.0.0:0"
+	transportAddr, err := tcp.AddressParser{}.Parse(addr)
+	ctx.NoError(err)
+
+	var routers []*Router
+
+	var routerIdx int
+	addRouter := func() *Router {
+		router := newRouterForTest(fmt.Sprintf("router-%03d", routerIdx), "", transportAddr, nil)
+		network.Routers.markConnected(router)
+		routerIdx++
+		return router
+	}
+
+	for i := 0; i < 50; i++ {
```

Comment:
this feels like a function you'll use numerous times when testing. forgot to make that comment on the other pr. this seems like a util test function type stuff, no?

Reply:
extracted these out, added to a test helper, put in util_test with stuff from debug.go